sdk/azstart/azstart.go

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package azstart
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/keyvault/azsecrets"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+)
+
+// UsingARMClients shows how to construct & use an ARM Client to invoke service methods
+func ExampleUsingARMClients() {
+	// Construct a credential type from the azidentity package
+	// or the module defining the client you wish to use
+	credential, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		panic(err)
+	}
+
+	// Construct an ARM client passing subscription ID, credential, & optional options
+	// This example creates a ResourceGroupsClient, but you can create any ARM client
+	client, err := armresources.NewResourceGroupsClient("<SubscriptionId>", credential, nil)
```

Comment:
We're introducing "client factories" in ARM modules.  Unsure if it's been merged for `armresources` yet, but perhaps we want to wait for that in order to promote it?

Reply:
Thanks @jhendrixMSFT for involving me. We are doing the refresh work to add client factory for all arm modules recently and will finish before the end of March. @sandeep-sen Could you wait for a while? I'll help to change the code after the new pattern could be adapted to all arm modules.

Reply:
@tadelesh I will wait till the client factory is released before I publish these code samples. I would need your assistance for updating this sample file